libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -597,6 +601,17 @@ class MapboxNavigation @VisibleForTesting internal constructor(
         return directionsSession.requestRoutes(routeOptions, routesRequestCallback)
     }
 
+    @ExperimentalPreviewMapboxNavigationAPI
+    suspend fun buildRouteOptions(
+        optionsBlock: (NoWaypointsOptionsBuilder) -> RouteOptionsBuilderWithWaypoints
+    ): RouteOptions {
+        tripSession.start(false)
```

Comment:
@tomaszrybakiewicz , what do you think we should do? Throw an exceptions if trip session hasn't been started yet?

Reply:
I agree with @tomaszrybakiewicz. As things stand, the session would never be started by itself, it's always initiated by developers, especially since it has billing consequences, next to functional consequences like starting all the observers.

Not only that but also the fact that we are not stopping the session that's started here, nor are we monitoring if it's been changed by developer which could result in the `buildRouteOptions` never returning because it would get stuck in `LocationFromTripSessionProvider`.

Let's start small and require the trip session to be started before someone calls `buildRouteOptions` or the proposed above `MapboxNavigation#requestRoutes(NavRouteOptionsBuilder)`. We'll still need to register a `TripSessionStateObserver` in the `LocationFromTripSessionProvider` to abort if the session was stopped before we got a location sample.

This will lest us experiment with the ergonomics without being blocked. As a next step, we can consider looking for a way to either bypass or initialize the session more ergonomically/silently when trying to obtain the location sample.

Reply:
Now I just throw an exception if trip session isn't started.

Reply:
> We'll still need to register a TripSessionStateObserver in the LocationFromTripSessionProvider to abort if the session was stopped before we got a location sample.

Added timeout for waining for location update. it's easier and handles more cases